### ISSUE TITLE
Attacking Neural Network ( keras model) for cifar10 dataset

### ISSUE THREAD

**Issue by SaiRaj07** [bethgelab/foolbox#197]
Here is my code snippet 

```
import foolbox 
import keras
import numpy as np
from keras import backend
from keras.models import load_model
from keras.datasets import cifar10
from keras.utils import np_utils
from foolbox.attacks import SaltAndPepperNoiseAttack
from foolbox.criteria import Misclassification
import matplotlib.pyplot as plt

########################################### Loading the model and preprocessing ###############################
backend.set_learning_phase(False)
model = keras.models.load_model('/home/path/Desktop/ADVERSARIAL ML/s and p/cifar_10.h5')
fmodel = foolbox.models.KerasModel(model, bounds=(-3,3))
(x_train, y_train), (images, labels) = cifar10.load_data()
x_train = x_train.astype('float32')
images = images.astype('float32')
mean = np.mean(x_train,axis=(0,1,2,3))
std = np.std(x_train,axis=(0,1,2,3))
x_train = (x_train-mean)/(std+1e-7)
images = (images-mean)/(std+1e-7)


######################################## Attacking all examples ################################################
adversarial_images=[]
for i in range(len(images)) :
    attack=foolbox.attacks.SaltAndPepperNoiseAttack(fmodel, criterion=Misclassification())
    img = images[i]
    label = labels[i]
    adversarial = attack(img, label)
    adversarial_images.append(adversarial)
```

I get the following error:
```
File "/home/path/anaconda3/lib/python3.6/site-packages/foolbox/adversarial.py", line 168, in __is_adversarial
    isinstance(is_adversarial, np.bool_)

AssertionError```

Note: Used similar model for mnist and there was no such error and attacks were generated successfully.

**Comment by wielandbrendel** [comment]
Could you print the full traceback? Also, can you get the labels from the foolbox model (fmodel.predictions(image))?

**Comment by wielandbrendel** [comment]
That's hard to tell without further details. Please try to supply a minimum working example that we can run and test.

**Comment by jonasrauber** [comment]
@SaiRaj07 Subtracting the mean and dividing by the standard deviation, are you sure the resulting images are between -3 and 3 as specified by your bounds. Even if they are, it's probably not a good idea to do it like this, because you tell the attack to operate in the space of inputs with values between -3 and 3, but there is no well-defined way to get back from that space to the image space. In general, multiplying by std and adding the mean will get you out of the image space.

What you should do instead is to pass proper images (e.g. with pixel values in (0, 1) or (0, 255)) to the attack. The preprocessing (subtracting mean, dividing by std) should be part of your model or should be specified using the `preprocessing` parameter of the Foolbox model wrapper (`KerasModel` in your case). Please have a look at the example in the Readme.

**Comment by gmt710** [comment]
adversarial = attack(img, label)
change to 
adversarial = attack(input_or_adv = img, label = label).

**Comment by jonasrauber** [comment]
@gmt710 whether you call them as positional arguments or keyword arguments surely should not make a difference ;-)

**Comment by gmt710** [comment]
@jonasrauber  Maybe,but by modifying this,it works.

**Comment by ksreddy-1994** [comment]
@jonasrauber @SaiRaj07 if you guys are still wondering, the issue might be that predict takes in 4 dimensions for the model (since its probably trained in batch or mini-batch) and thus returns an array of bools instead of a single bool. FYI

**Comment by JiweiTian** [comment]
Have you fixed this problem? How?

**Comment by JiweiTian** [comment]
@SaiRaj07

**Comment by nini140** [comment]
@SaiRaj07 You can try
labels = labels.reshape(-1)
it works for me.
